modules/react/checkbox/lib/Checkbox.tsx

```diff
@@ -330,18 +339,18 @@ export const Checkbox = createComponent('input')({
             checked={checked}
             disabled={disabled}
             id={inputId}
-            ref={ref}
+            ref={localRef}
             type="checkbox"
             variant={variant}
             aria-checked={indeterminate ? 'mixed' : checked}
             {...elemProps}
           />
           <CheckboxRipple variant={variant} />
           <CheckboxBackground variant={variant} checked={checked} disabled={disabled}>
-            <CheckboxCheck checked={checked}>
+            <CheckboxCheck checked={checked} indeterminate={indeterminate}>
               {indeterminate ? (
                 <IndeterminateBox variant={variant} />
-              ) : (
+              ) : checked ? (
```

Comment:
This fixes a different sneaky bug where the checkbox flashes the checked icon when moving between indeterminate true to false

Reply:
Nice catch!